examples/chardev.c

```diff
@@ -111,10 +109,15 @@ static ssize_t device_read(struct file *filp, /* see include/linux/fs.h   */
 {
     /* Number of bytes actually written to the buffer */
     int bytes_read = 0;
+    const char *msg_ptr = msg;
 
-    /* If we are at the end of message, return 0 signifying end of file. */
-    if (*msg_ptr == 0)
+    /* If we are at the end of message, reset the offset and return 0 (which
```

Comment:
Rewrite as following:
```c
    if (!*(msg_ptr + *offset)) { /* we are at the end of message */
        *offset = 0; /* reset the offset */
        return 0; /* signify end of file */
    }
```

For writing purpose, we would stick to shorter code snip.